autoarray/plot/wrap/base/figure.py

```diff
@@ -0,0 +1,80 @@
+import matplotlib.pyplot as plt
+from typing import Union, Tuple
+
+from autoarray.plot.wrap.base.abstract import AbstractMatWrap
+
+
+class Figure(AbstractMatWrap):
+    """
+    Sets up the Matplotlib figure before plotting (this is used when plotting individual figures and subplots).
+
+    This object wraps the following Matplotlib methods:
+
+    - plt.figure: https://matplotlib.org/3.3.2/api/_as_gen/matplotlib.pyplot.figure.html
+    - plt.close: https://matplotlib.org/3.3.2/api/_as_gen/matplotlib.pyplot.close.html
+
+    It also controls the aspect ratio of the figure plotted.
+    """
+
+    @property
+    def config_dict(self):
+        """
+        Creates a config dict of valid inputs of the method `plt.figure` from the object's config_dict.
+        """
+
+        config_dict = super().config_dict
+
+        if config_dict["figsize"] == "auto":
+            config_dict["figsize"] = None
+        elif isinstance(config_dict["figsize"], str):
+            config_dict["figsize"] = tuple(
+                map(int, config_dict["figsize"][1:-1].split(","))
+            )
+
+        return config_dict
+
+    def aspect_for_subplot_from(self, extent):
+
+        ratio = float((extent[1] - extent[0]) / (extent[3] - extent[2]))
+
+        if self.config_dict["aspect"] in "square":
+            return ratio
+        elif self.config_dict["aspect"] in "auto":
+            return 1.0 / ratio
+        elif self.config_dict["aspect"] in "equal":

```

Comment:
Assign aspect as a variable to avoid repetition.

Raise a ValueError if aspect does not match one of the excepted values.

Alternatively you could use an Enum:

```python
class Aspect(Enum):
    square = 1
    auto = 2
    equal = 3

aspect = Aspect[self.config_dict["aspect"]]

if aspect == Aspect.square:
    ...